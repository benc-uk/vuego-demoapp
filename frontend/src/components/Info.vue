<!--
Info page, fetches system info data from API
--------------------------------------------
Ben C, April 2018, Updated for Vue3 2021
-->

<template>
  <div>
    <div v-if="error" class="alert text-dark alert-warning">
      <h4>There was a problem üò•</h4>
      <div class="errmsg">
        {{ error }}
      </div>
    </div>

    <spinner v-if="!info && !error" />

    <div v-if="info" class="card border-light bg-dark mb-3 rounded-3">
      <h4 class="card-header bg-info"><i class="fas fa-cogs fa-fw me-3"></i> System Information</h4>
      <div class="card-body">
        <table class="table table-hover table-striped fs-5">
          <tbody>
            <tr class="table-dark">
              <td><i class="fab fa-docker fa-fw me-1"></i> Containerized</td>
              <td>{{ info.isContainer ? 'Running in a container! üòÅ' : 'Not running in a container üò¢' }}</td>
            </tr>
            <tr class="table-dark">
              <td><i class="fas fa-dharmachakra fa-fw me-1"></i> Kubernetes</td>
              <td>{{ info.isKubernetes ? 'Running in Kubernetes! üòÑ' : 'Not running in Kubernetes üò™' }}</td>
            </tr>
            <tr class="table-dark">
              <td><i class="fas fa-home fa-fw me-1"></i> Hostname</td>
              <td>{{ info.hostname }}</td>
            </tr>
            <tr class="table-dark">
              <td><i class="fas fa-wrench fa-fw me-1"></i> Platform</td>
              <td>{{ info.platform }} {{ info.architecture }}</td>
            </tr>
            <tr class="table-dark">
              <td><i class="fas fa-laptop-code fa-fw me-1"></i> Operating System</td>
              <td>{{ info.os }}</td>
            </tr>
            <tr class="table-dark">
              <td><i class="fas fa-microchip fa-fw me-1"></i> Processors</td>
              <td>{{ info.cpuCount }} x {{ info.cpuModel }}</td>
            </tr>
            <tr class="table-dark">
              <td><i class="fas fa-memory fa-fw me-1"></i> Memory</td>
              <td>{{ (info.mem / (1024 * 1024 * 1024)).toFixed(2) }} GB</td>
            </tr>
            <tr class="table-dark">
              <td><i class="fas fa-flask fa-fw me-1"></i> Go Version</td>
              <td>{{ info.goVersion }}</td>
            </tr>
            <tr class="table-dark">
              <td><i class="fas fa-project-diagram fa-fw me-1"></i> Network Address</td>
              <td>{{ info.netHost }}</td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>

    <br />

    <div v-if="envVars" class="card border-light bg-dark mb-3 rounded-3">
      <h4 class="card-header bg-info"><i class="fas fa-comment-dollar fa-fw me-3"> </i> Environment Variables</h4>
      <div class="card-body">
        <table class="vartable">
          <tbody>
            <tr v-for="(envVar, index) in envVars" :key="index">
              <td>{{ envVar.name }}</td>
              <td class="value">
                {{ envVar.value }}
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</template>

<script>
import apiMixin from '../mixins/apiMixin.js'
import Spinner from './Spinner.vue'
const info = null

export default {
  components: {
    Spinner
  },

  // Adds functions to call the API
  mixins: [apiMixin],

  data: function () {
    return {
      info: info,
      error: null
    }
  },

  computed: {
    envVars: function () {
      let vars = []
      if (!this.info) return null
      for (const envVar of this.info.envVars) {
        const parts = envVar.split('=')
        const name = parts[0]
        const value = parts[1]
        if (name.includes('PATH')) {
          continue
        }
        if (name.includes('NPM_')) {
          continue
        }
        if (name.includes('VSCODE_')) {
          continue
        }
        vars.push({ name, value })
      }
      vars = vars.sort((e1, e2) => {
        if (e1.name > e2.name) {
          return 1
        }
        return -1
      })
      return vars
    }
  },

  created() {
    this.getInfo()
  },

  methods: {
    getInfo: async function () {
      try {
        this.info = await this.apiGetInfo()
      } catch (err) {
        this.error = err
      }
    }
  }
}
</script>

<style scoped>
.vartable {
  font-size: 19px;
  background-color: #141414;
  border-top: none;
  border-bottom: none;
  color: #afafaf;
  overflow-x: scroll;
  width: 100%;
  table-layout: fixed;
  border-radius: 0.3rem;
}

.vartable tr {
  border-top: solid 1px #444444;
  font-family: 'Consolas', 'Courier New', Courier, monospace;
}
.vartable tr:hover {
  background-color: #222222;
}
.vartable tr:first-child {
  border-top: none;
}
.vartable td {
  padding: 0.5rem;
}

.value {
  color: #2dc22d;
}
</style>
