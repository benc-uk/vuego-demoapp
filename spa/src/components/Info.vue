<!--
Info page, fetches system info data from API
--------------------------------------------
Ben C, April 2018
-->

<template>
  <div>
    <b-alert v-if="error" show variant="warning">
      <h4>There was a problem üò•</h4>
      <div class="errmsg">
        {{ error }}
      </div>
    </b-alert>

    <spinner v-if="!info && !error" />

    <div class="card">
      <div class="card-header bg-success h3"><fa icon="cogs" />&nbsp; System Information</div>
      <div class="card-body">
        <table v-if="info" class="table table-hover table-sm table-striped">
          <tbody>
            <tr>
              <td><fa :icon="['fab', 'docker']" fixed-width /> Containerized</td>
              <td>{{ info.isContainer ? 'Running in a container! üòÅ' : 'Not running in a container üò¢' }}</td>
            </tr>
            <tr>
              <td><fa icon="dharmachakra" fixed-width /> Kubernetes</td>
              <td>{{ info.isKubernetes ? 'Running in Kubernetes! üòÑ' : 'Not running in Kubernetes üò™' }}</td>
            </tr>
            <tr>
              <td><fa icon="home" fixed-width /> Hostname</td>
              <td>{{ info.hostname }}</td>
            </tr>
            <tr>
              <td><fa icon="wrench" fixed-width /> Platform</td>
              <td>{{ info.platform }} {{ info.architecture }}</td>
            </tr>
            <tr>
              <td><fa icon="laptop-code" fixed-width /> Operating System</td>
              <td>{{ info.os }}</td>
            </tr>
            <tr>
              <td><fa icon="microchip" fixed-width /> Processors</td>
              <td>{{ info.cpuCount }} x {{ info.cpuModel }}</td>
            </tr>
            <tr>
              <td><fa icon="memory" fixed-width /> Memory</td>
              <td>{{ (info.mem / (1024 * 1024 * 1024)).toFixed(2) }} GB</td>
            </tr>
            <tr>
              <td><fa icon="flask" fixed-width /> Go Version</td>
              <td>{{ info.goVersion }}</td>
            </tr>
            <tr>
              <td><fa icon="project-diagram" fixed-width /> Network Address</td>
              <td>{{ info.netHost }}</td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>

    <br />

    <div class="card">
      <div class="card-header bg-success h3"><fa icon="comment-dollar" />&nbsp; Environment Variables</div>
      <div class="card-body">
        <table v-if="info" class="vartable">
          <tbody>
            <tr v-for="(envVar, index) in envVars" :key="index">
              <td>{{ envVar.name }}</td>
              <td class="value">
                {{ envVar.value }}
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</template>

<script>
import apiMixin from '../mixins/apiMixin.js'
import Spinner from './Spinner.vue'
const info = null

export default {
  components: {
    Spinner
  },

  // Adds functions to call the API
  mixins: [apiMixin],

  data: function() {
    return {
      info: info,
      error: null
    }
  },

  computed: {
    envVars: function() {
      let vars = []
      for (const envVar of this.info.envVars) {
        const parts = envVar.split('=')
        const name = parts[0]
        const value = parts[1]
        if (name.includes('PATH')) {
          continue
        }
        if (name.includes('NPM_')) {
          continue
        }
        if (name.includes('VSCODE_')) {
          continue
        }
        vars.push({ name, value })
      }
      vars = vars.sort((e1, e2) => {
        if (e1.name > e2.name) {
          return 1
        }
        return -1
      })
      return vars
    }
  },

  created() {
    this.getInfo()
  },

  methods: {
    getInfo: async function() {
      try {
        this.info = await this.apiGetInfo()
      } catch (err) {
        this.error = err
      }
    }
  }
}
</script>

<style scoped>
.vartable {
  font-size: 19px;
  background-color: #141414;
  border-top: none;
  border-bottom: none;
  color: #afafaf;
  overflow-x: scroll;
  width: 100%;
  table-layout: fixed;
  border-radius: 0.3rem;
}

.vartable tr {
  border-top: solid 1px #444444;
  font-family: 'Consolas', 'Courier New', Courier, monospace;
}
.vartable tr:hover {
  background-color: #222222;
}
.vartable tr:first-child {
  border-top: none;
}
.vartable td {
  padding: 0.5rem;
}

.value {
  color: #2dc22d;
}
</style>
