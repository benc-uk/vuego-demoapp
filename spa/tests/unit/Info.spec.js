import { createLocalVue, shallowMount } from '@vue/test-utils'
import Info from '@/components/Info.vue'
import flushPromises from 'flush-promises'

const localVue = createLocalVue()

const mockMixin = {
  methods: {
    apiGetInfo() {
      return new Promise((resolve) => {
        resolve({
          hostname: 'test',
          isContainer: true,
          isKubernetes: true,
          platform: 'PDP-11',
          architecture: '98 bit',
          os: 'MegaOS: 3000',
          mem: 1234567890,
          envVars: ['UNIT_TESTS=Are pointless']
        })
      })
    }
  }
}

describe('Info.vue', () => {
  it('renders info screen', async () => {
    const wrapper = shallowMount(Info, { localVue, mixins: [mockMixin] })
    await flushPromises()
    expect(wrapper).toMatchSnapshot()
  })
})
