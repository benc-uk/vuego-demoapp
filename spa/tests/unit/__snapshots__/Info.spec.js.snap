// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info.vue renders info screen 1`] = `
<div>
  <!---->
  <!---->
  <div class="card">
    <div class="card-header bg-success h3">
      <fa icon="cogs"></fa>&nbsp; System Information
    </div>
    <div class="card-body">
      <table class="table table-hover table-sm table-striped">
        <tbody>
          <tr>
            <td>
              <fa icon="fab,docker" fixed-width=""></fa> Containerized
            </td>
            <td>Running in a container! üòÅ</td>
          </tr>
          <tr>
            <td>
              <fa icon="dharmachakra" fixed-width=""></fa> Kubernetes
            </td>
            <td>Running in Kubernetes! üòÑ</td>
          </tr>
          <tr>
            <td>
              <fa icon="home" fixed-width=""></fa> Hostname
            </td>
            <td>test</td>
          </tr>
          <tr>
            <td>
              <fa icon="wrench" fixed-width=""></fa> Platform
            </td>
            <td>PDP-11 98 bit</td>
          </tr>
          <tr>
            <td>
              <fa icon="laptop-code" fixed-width=""></fa> Operating System
            </td>
            <td>MegaOS: 3000</td>
          </tr>
          <tr>
            <td>
              <fa icon="microchip" fixed-width=""></fa> Processors
            </td>
            <td> x </td>
          </tr>
          <tr>
            <td>
              <fa icon="memory" fixed-width=""></fa> Memory
            </td>
            <td>1.15 GB</td>
          </tr>
          <tr>
            <td>
              <fa icon="flask" fixed-width=""></fa> Go Version
            </td>
            <td></td>
          </tr>
          <tr>
            <td>
              <fa icon="project-diagram" fixed-width=""></fa> Network Address
            </td>
            <td></td>
          </tr>
        </tbody>
      </table>
    </div>
  </div> <br>
  <div class="card">
    <div class="card-header bg-success h3">
      <fa icon="comment-dollar"></fa>&nbsp; Environment Variables
    </div>
    <div class="card-body">
      <table class="vartable">
        <tbody>
          <tr>
            <td>UNIT_TESTS</td>
            <td class="value">
              Are pointless
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
